src/llama-graph.h

```diff
@@ -594,12 +597,13 @@ struct llm_graph_context {
     //
 
     ggml_tensor * build_recurrent_state(
-             ggml_cgraph * gf,
-             ggml_tensor * s,
-             ggml_tensor * state_copy,
-                 int32_t   state_size,
-                 int32_t   n_seqs,
-                    bool   avoid_copies = false) const;
+                                 ggml_cgraph * gf,
+                                 ggml_tensor * s,
+                                 ggml_tensor * state_copy,
+                                     int32_t   state_size,
+                                     int32_t   n_seqs,
+                                        bool   avoid_copies = false,
+        const llama_kv_cache_recurrent_state * kv_state = nullptr) const;
```

Comment:
NIT: This was the closest I could come up to for consistent indentation, but it's... ugly? Very open to preferred style suggestions.

Non-NIT: I don't love adding this, especially after `avoid_copies` since it now forces all layer builders that may be in a hybrid architecture to explicitly set `avoid_copies`.